### PR TITLE
feat: automatically enable allowHTTP1 by default with http2 & hmr

### PR DESCRIPTION
**Pull request by illusionalsagacity**
This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] tests
- [ ] documentation
- [ ] metadata

### Breaking Changes?

- [x] yes
- [ ] no

If yes, please describe the breakage.

Changes the default configuration for the http2 server if `hmr` is true.

### Please Describe Your Changes

relates to #183 and #184 

I added the `tls-keygen` dev dependency in order to generate the keys, unfortunately browsers do not seem to support unencrypted http2 at this time, even on localhost. Probably worth some followup work to make the http2 option require some of the `http2.createSecureServer` options to avoid any footguns?

### CODE REVIEW

**Review by PlayMa256 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ If `true`, will enable [`Hot Module Replacement`](https://webpack.js.org/concept
 
 _Note: If the build process generates errors, the client (browser) will not be notified of new changes and no HMR will be performed. Errors must be resolved before HMR can proceed._
 
-_Note: If using in combination with `http2`, the `http2` option `allowHTTP1` must be enabled for the HMR WS connection to work._
+_Note: If using in combination with `http2`, the `http2` option `allowHTTP1` will enabled by default for the HMR WS connection to work._
```
Comment (shellscape) [inline README.md:141]:
```suggestion
_Note: If using in combination with `http2`, the `http2` option `allowHTTP1` will be enabled by default for the HMR WS connection to work._
```

**Review by shellscape — CHANGES_REQUESTED** [review]
I see the new test fixtures added, but not any tests. Is that pending or uncommitted?

File: README.md (hunk 1 above)
Comment (illusionalsagacity) [reply README.md]: So I can't get the integration tests to run to completion locally, particularly the multi-hmr one seems to be troublesome.  It does happen as well on the master branch, so I'm not quite sure what is going on there.

Comment (bebraw) [reply README.md]: Can you tell more about the test issue? How does it fail exactly?

**Comment by shellscape** [comment]
@illusionalsagacity I'll try and step into your branch to get the tests work. probably later this week.

**Comment by illusionalsagacity** [comment]
> @illusionalsagacity I'll try and step into your branch to get the tests work. probably later this week.

Alright, thanks. To answer @bebraw the issue I'm seeing is that chromium just sits on the `about:blank` page and the tests hang for me. Probably related to some of the changes I made with the puppeteer config, I would assume.

**Pull request closed without merging**